File: sample-java/src/main/java/com/smart/github/app/trend/TrendActivity.java
package com.smart.github.app.trend;

import android.view.MenuItem;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import com.compact.app.CompactActivity;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.smart.github.R;
import io.reactivex.android.schedulers.AndroidSchedulers;

import javax.inject.Inject;

public class TrendActivity extends CompactActivity {

    @Inject
    ViewModelProvider.Factory factory;

    @BindView(R.id.message)
    TextView message;
    @BindView(R.id.nav_view)
    BottomNavigationView navView;
    @BindView(R.id.list)
    RecyclerView recyclerView;

    private TrendViewModel viewModel;
    private TrendAdapter adapter;
    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    message.setText(R.string.title_home);
                    return true;
                case R.id.navigation_dashboard:
                    message.setText(R.string.title_dashboard);
                    return true;
                case R.id.navigation_notifications:
                    message.setText(R.string.title_notifications);
                    return true;
            }
            return false;
        }
    };

    @Override
    protected int layoutRes() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate() {
        viewModel = ViewModelProviders.of(this, factory).get(TrendViewModel.class);

        adapter = new TrendAdapter();
        recyclerView.setAdapter(adapter);

        viewModel.get()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(it -> adapter.swap(it));

        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }
}

File: sample/src/main/java/com/smart/github/app/trend/TrendActivity.kt
package com.smart.github.ui.trend

import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.RecyclerView
import com.compact.app.CompactActivity
import com.smart.github.R
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.functions.Consumer
import javax.inject.Inject

class TrendActivity : CompactActivity() {

    @Inject
    lateinit var factory: ViewModelProvider.Factory

    private lateinit var viewModel: TrendViewModel

    override fun layoutRes(): Int {
        return R.layout.activity_main
    }

    override fun onCreate() {
        viewModel = ViewModelProviders.of(this, factory).get(TrendViewModel::class.java)

        var recyclerView = findViewById<RecyclerView>(R.id.list)
        val adapter = TrendAdapter().also {
            recyclerView.adapter = it
        }

        viewModel.get()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(Consumer { adapter.swap(it) })
    }
}
